tox.ini

```diff
@@ -6,11 +6,20 @@ skipsdist = True
 usedevelop = True
 deps =
   -r{toxinidir}/test-requirements.txt
-  -r{toxinidir}/requirements.txt
+commands = nosetests --tests tests/unit
 
-commands = python setup.py nosetests
-
-install_command=pip install --process-dependency-links {opts} {packages}
+[testenv:integration]
+deps =
+  -r{toxinidir}/test-requirements.txt
+whitelist_externals =
+    docker
+commands_pre =
+    docker build . -t fake-ubersmith-integration
+    docker run -itd --rm -p 9131:9131 --name fake-ubersmith fake-ubersmith-integration
```

Comment:
Why not using the environment variable "APP_EXPOSED_PORT" defined in  the Dockerfile for the port ? something like : {env:APP_EXPOSED_PORT}  -> https://tox.readthedocs.io/en/latest/config.html#environment-variable-substitutions

Reply:
Because it is actually defined in the dockerfile and thus, not available elsewhere.  there should probably be a refactor of the whole port management somewhere but that should be for another time.   at the moment port is also hardcoded in the tests that a ran by tox (see the tests/integration/base.py)